### ISSUE TITLE
Respect font scaling

### ISSUE THREAD

**Issue by Vorror** [framework7io/framework7#2441]
Framework7 does not currently respect the ios font scaling.

### iOS with default font scaling

![img_1220](https://user-images.githubusercontent.com/3613230/41740930-92c0377c-7567-11e8-8343-7c5a0d9f9eec.jpg)
![img_1217](https://user-images.githubusercontent.com/3613230/41740933-97d0d5c8-7567-11e8-9f95-d789cdb20abb.jpg)
![img_1218](https://user-images.githubusercontent.com/3613230/41740937-9a652348-7567-11e8-8e32-97b6c6f15d8c.jpg)
![img_1219](https://user-images.githubusercontent.com/3613230/41740940-9cf119a0-7567-11e8-9291-0a8e94498154.jpg)

### iOS with large font scaling(not working)

![img_1221](https://user-images.githubusercontent.com/3613230/41740951-a5cf3b60-7567-11e8-8000-04c43c0879de.jpg)
![img_1222](https://user-images.githubusercontent.com/3613230/41740957-abfc140e-7567-11e8-8aaf-53fc1b4d4283.jpg)
![img_1223](https://user-images.githubusercontent.com/3613230/41740973-b5441e6c-7567-11e8-8dbb-9ff5178ee1c1.jpg)

### Background

We have a lot of user's who don't use the default iOS font scaling.
Here's some tech info in regards to supporting scalable fonts: http://www.interactiveaccessibility.com/blog/text-resizing-web-pages-ios-using-dynamic-type

I can confirm using their test page (http://ia11y.github.io/Coding-Patterns/iOS/text-resizing/resizable-vs-non-resizable-font-ios.html) that the font does scale on iOS device.

Basically this would require switch to em font sizes. I see there's a v3 of framework7 in the works. Maybe this could be a feature for that version?

**Comment by Vorror** [comment]
For the most part F7 could simply convert px -> em using a less mixin. Although any parent elements that have hard coded widths/heights would need to be fixed by doing something reasonable(truncating text, line-breaking, etc)

**Comment by nolimits4web** [comment]
Rewriting all font sizes to `em` doesn't seem a good option for me. Will take a look what can be done here

**Comment by nolimits4web** [comment]
This issue was incorrectly closed on 2026-05-19 by a compromised credential (Shai-Hulud npm supply-chain worm). Reopening — apologies for the noise.
